mentions/src/main/kotlin/io/spine/examples/pingh/mentions/GitHubClientServiceImpl.kt

```diff
@@ -0,0 +1,179 @@
+                findMentions(username, token, ItemType.PULL_REQUEST)
+
+    /**
+     * Requests GitHub for mentions of a user in issues or pull requests,
+     * then looks for where the user was specifically mentioned in that item.
+     */
+    @Throws(CannotFetchMentionsFromGitHubException::class)
+    private fun findMentions(
+        username: Username,
+        token: PersonalAccessToken,
+        itemType: ItemType
+    ): Set<Mention> {
+        val mentions = mutableSetOf<Mention>()
+        val userTag = "@${username.value}"
+        val items = searchIssuesOrPullRequests(username, token, itemType)
+
+        for (item in items.itemList) {
+            if (item.body.contains(userTag)) {
+                mentions.add(Mention::class.buildFromFragment(item))
+            }
+            val comments = getComments(item.commentsUrl, token)
+            for (comment in comments.itemList) {
+                if (comment.body.contains(userTag)) {
+                    mentions.add(Mention::class.buildFromFragment(comment, item.title))
+                }
+            }
+        }
+        return mentions.toSet()
+    }
+
+    /**
+     * Requests to GitHub for searching issues or pull requests that mention a particular user.
+     *
+     * There is only one endpoint for searching issues or pull requests from the GitHub API,
+     * but when using GitHub Apps, it is mandatory to specify what is being searched
+     * for using the `is:` filter. Therefore, the type of item being searched for is specified.
+     *
+     * @see <a href="https://docs.github.com/en/rest/search/search#search-issues-and-pull-requests">
+     *     Search issues and pull requests</a>
+     */
+    @Throws(CannotFetchMentionsFromGitHubException::class)
+    private fun searchIssuesOrPullRequests(
+        username: Username,
+        token: PersonalAccessToken,
+        itemType: ItemType
+    ): IssuesAndPullRequestsSearchResult =
+        runBlocking {
+            val response = client.get("https://api.github.com/search/issues") {
+                url {
+                    parameters.append(
+                        "q",
+                        "is:${itemType.gitHubName()} mentions:${username.value}"
+                    )
+                    parameters.append("per_page", "100")
+                    parameters.append("sort", "updated")
+                    parameters.append("order", "desc")
+                }
+                configureHeaders(token)
+            }
+            if (response.status != HttpStatusCode.OK) {
+                throw CannotFetchMentionsFromGitHubException(response.status.value)
+            }
+            parseIssuesAndPullRequestsFromJson(response.body())
+        }
+
+    /**
+     * Requests comments from GitHub on their URL previously received.
+     */
+    @Throws(CannotFetchMentionsFromGitHubException::class)
+    private fun getComments(url: String, token: PersonalAccessToken): CommentsGetResult =
```

Comment:
Please refrain from naming anything starting with `get`.

The reason is that `getSomething()` has always been a way to obtain something we already have. So this method executed at no cost. In 99.9% of your scenarios, you won't need POJO-style getters (as you are in Kotlin). And the methods you create, again, won't just return the existing value, but rather perform some job to retrieve the value. Hence, they aren't "getters".